### PR TITLE
Add java.configuration.maven.defaultMojoExecutionAction

### PR DESCRIPTION
**Pull request by snjeza**
Related issues:
- https://github.com/redhat-developer/vscode-java/issues/2889
- https://github.com/eclipse-m2e/m2e-core/issues/1013

Requires https://github.com/eclipse/eclipse.jdt.ls/pull/2426

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,6 +224,7 @@ The following settings are supported:
 
 New in 1.15.0
 * `java.import.maven.disableTestClasspathFlag` : Enable/disable test classpath segregation. When enabled, this permits the usage of test resources within a Maven project as dependencies within the compile scope of other projects. Defaults to `false`.
+* `java.configuration.maven.defaultMojoExecutionAction` : Specifies default mojo execution action. Defaults to `ignore`.
```
Comment (rgrunber) [inline README.md:227]: Does it make sense to change this to : 

Specifies default mojo execution action when no associated metadata can be detected. Defaults to `ignore`.

Ie. this is specifically about mojos that m2e doesn't recognize, but we don't need to go into those details. If it makes sense to you, feel free to make the change.

File: package.json [hunk 2]
```diff
@@ -404,6 +404,18 @@
           "description": "Specifies severity if the plugin execution is not covered by Maven build lifecycle.",
           "scope": "window"
         },
+        "java.configuration.maven.defaultMojoExecutionAction": {
+          "type": "string",
+          "enum": [
+            "ignore",
+            "execute",
```
Comment (rgrunber) [inline package.json:411]: Does it make sense to put "execute" at the bottom since the sorting seems to be by least amount of disruption/change to most ?

**Review by rgrunber — COMMENTED** [review]
Looks fine to me. Just some minor changes might be needed.

File: README.md (hunk 1 above)
Comment (snjeza) [reply README.md]: Fixed.

File: package.json (hunk 2 above)
Comment (snjeza) [reply package.json]: Fixed.

**Comment by snjeza** [comment]
@rgrunber I have updated the PR.

**Review by rgrunber — APPROVED** [review]
I think `ignore` was fine. Is there a reason to remove the back-ticks ? Change is fine so I'd just add them back and feel free to merge.

**Pull request merged**